flake.nix

```
@@ -0,0 +1,125 @@
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+
+    in
+    {
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          inherit (pkgs) stdenv lib ;
+        in
+        {
+          ${pname} = pkgs.buildGo121Module rec {
+            inherit pname self;
+            version = lib.fileContents ./VERSION;
+            gitCommit = if (self ? rev) then self.rev else self.dirtyRev;
+            state = if (self ? rev) then "clean" else "dirty";
+
+            # This vendorHash represents a dervative of all go.mod dependancies and needs to be adjusted with every change
+            vendorHash = "sha256-CA3p9QNHo7mHCoXkuOojFAJen3TdieSsoQVivxPk3yw=";
+
+            src = ./.;
+
+            ldflags = [
+              "-s" "-w"
+              "-X github.com/open-component-model/ocm/pkg/version.gitVersion=${version}"
+              "-X github.com/open-component-model/ocm/pkg/version.gitTreeState=${state}"
+              "-X github.com/open-component-model/ocm/pkg/version.gitCommit=${gitCommit}"
+            # "-X github.com/open-component-model/ocm/pkg/version.buildDate=1970-01-01T0:00:00+0000"
+            ];
+
+            CGO_ENABLED = 0;
+
+            subPackages = [ 
+              "cmds/ocm" 
+              "cmds/helminstaller"
+              "cmds/demoplugin"
+              "cmds/ecrplugin"
+            ];
+
+            nativeBuildInputs = [ pkgs.installShellFiles ];
+
+            postInstall = ''
+              installShellCompletion --cmd ${pname} --zsh  <($out/bin/${pname} completion zsh)
+              installShellCompletion --cmd ${pname} --bash <($out/bin/${pname} completion bash)
+              installShellCompletion --cmd ${pname} --fish <($out/bin/${pname} completion fish)
+            '';
+
+            meta = with lib; {
+              description = "Open Component Model (OCM) is an open standard to describe software bills of delivery (SBOD)";
+              longDescription = ''
+                The Open Component Model (OCM) is an open standard to describe software bills of delivery (SBOD). OCM is a technology-agnostic and machine-readable format focused on the software artifacts that must be delivered for software products.
+              '';
+              homepage = "https://ocm.software";
+              license = licenses.asl20;
+              platforms = supportedSystems;
+            };
+          };
+        });
+      
+      # Add dependencies that are only needed for development
+      devShells = forAllSystems (system:
+        let 
+          pkgs = nixpkgsFor.${system};
+        in
+        {
+          default = pkgs.mkShell {
+            buildInputs = with pkgs; [ 
+              go_1_21 
+              gopls 
+              gotools 
+              go-tools
```

```suggestion
              gotools # goimports
              go-tools # staticcheck
```
I would recommend to document what contains what 😅